.github/workflows/templating.yml

```
@@ -0,0 +1,28 @@
+name: Templating
+on:
+  schedule:
+    # Run every Monday at 00:00 UTC
+    - cron: "0 0 * * 1"
+  pull_request:
+    paths:
+      - ".github/workflows/templating.yml"
+  workflow_dispatch: # Allow manual triggering
+
+jobs:
+  render-template:
+    name: Render template
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      pull-requests: write
+    steps:
+      - id: create-token
+        uses: actions/create-github-app-token@v1
+        with:
+          app-id: ${{ secrets.APP_ID }}
+          private-key: ${{ secrets.APP_PRIVATE_KEY }}
+      - uses: munich-quantum-toolkit/templates@main
+        with:
+          token: ${{ steps.create-token.outputs.token }}
+          organization: munich-quantum-toolkit
+          repository: core
```

I believe we can even make this workflow a step more generic so that it can also be more easily templated.
```suggestion
          organization: ${{ github.repository_owner }}
          repository: ${{ github.event.repository.name }}
```